proto/payload.proto

```diff
@@ -0,0 +1,48 @@
+syntax = "proto3";
+
+import public "proto/utility.proto";
+
+package payload;
+
+enum PayloadType {
+    ACKNOWLEDGEMENT = 0;
+    REGISTRATION = 1;
+    DISCOVERY_REQUEST = 2;
+    DISCOVERY_RESPONSE = 3;
+    TASK_REQUEST = 4;
+    TASK_RESPONSE = 5;
+}
+
+message PayloadMessage {
+    string id = 1;
+    string senderId = 2;
+    bytes senderIpAddress = 3;
+    PayloadType payloadType = 4;
```

Comment:
Just a thought: might be helpful to add a timestamp field to our messages transferred over the network. 

This field might be a better fit in the actual message.h that is sent rather than the payload though.

Reply:
Fair point. `PayloadMessage` would be the actual message.h that is sent between computers. I had attempted to rename this to `Message`, but that was buggy since protobuf has already defined a `Message` class. Worth renaming?

Reply:
I would say we should add the protobuf timestamp as a separate PR task, as that involves some other C++ dependency work

Reply:
Yea sounds great. I agree we don't have to make any changes for this PR.